src/errors.cpp

```diff
@@ -19,15 +19,27 @@ struct SqlStateMapping
 
 static const struct SqlStateMapping sql_state_mapping[] =
 {
+    { "01002", 5, &OperationalError },
+    { "08001", 5, &OperationalError },
+    { "08003", 5, &OperationalError },
```

Comment:
08003 is "Connection not open".  Wouldn't that be ProgrammingError?

Reply:
When we've had it, it has been the result of executing a query on a connection that has unexpectedly closed at the server end (eg due to a timeout), which feels more like an operational problem.

(Although, it's unclear whether this is what the sqlstate is *supposed* to represent, as the definitions are pretty fuzzy and don't necessarily match reality...)

Reply:
Well then, let's leave them as-is.  Thanks.